Section header fix (#465)

VegetarianZombie opened the pull request:
This pull request resolves the issue with additional lines rendering in iOS 14. This was due to list sections being nested inside of each other. This pull request refactors the sections inside of a group that resolves the issue. It also sets the width of the list cell to the width of the device.
 
Before:

![Screen Shot 2020-09-14 at 2 53 04 PM](https://user-images.githubusercontent.com/973751/93126180-0ce5ba80-f69a-11ea-9f1c-4e9b8fc75609.png)

After:

![Screen Shot 2020-09-14 at 2 41 32 PM](https://user-images.githubusercontent.com/973751/93125756-934dcc80-f699-11ea-82a9-0fc85572e0ae.png)

Note: The project contains a Group struct that causes a name collision with the SwiftUI Group.

sammyd reviewed (commented):
This looks good to me. Since it requires Xcode 12, I want to wait until we can switch CI to merge it. Hopefully it'll be in the next day or so.

sammyd commented:
@VegetarianZombie actually, how come the styling of the header has changed? I don't think it should be centred / capitalised.

VegetarianZombie commented:
Hi Sam - Okay, that was a mistake on my part. I'll update the PR to use the correct styling information.

sammyd commented on Emitron/Emitron/UI/Shared/Content Detail/CourseHeaderView.swift:
```diff
@@ -36,11 +37,8 @@ struct CourseHeaderView: View {
       Text(name)
```
Not sure why, but this is now centred and capitals?

sammyd reviewed (changes requested):
As specified above—not sure why the typography has changed...

VegetarianZombie committed:
left justify section headers

VegetarianZombie commented:
Hi @sammydl, I investigated the issue today. The casing change is a result from iOS 14. By default, section headers appear now to be uppercased. This can be changed with the 'textCase' modifier.

I updated the pull request to left justify the header. I also added a conditional statement to set the text case on iOS 14 and above. Pre iOS 14 is fine.

sammyd commented:
Thanks @VegetarianZombie. I'll get this rebased, resolve the merge conflicts, and get it merged.

sammyd committed:
Update project settings to Xcode 12 beta 5, fixes warning

sammyd committed:
Remove section nesting to remove additional line

sammyd committed:
Add a conditional to set the casing on the section header

sammyd committed:
Fixing my merge-conflict resolution screw-up

sammyd reviewed (approved):
:+1: I've rebased it, and will go ahead and merge it once CI has completed.

Nice work—thanks for this @VegetarianZombie

sammyd commented:
This fixed #465
